implementation.md

```diff
@@ -0,0 +1,163 @@
+# Implementation Notes
+
+HyperApp's design is based on the [Elm Architecture](https://guide.elm-lang.org/architecture/) which enables you to create scalable browser-based applications using a functional paradigm.
+
+The logic of every program using this architecture will break into three parts :
+
+* **Model** - It represents the state of your application
+* **Update** - A way to update your state (actions)
+* **View** - View your state as HTML
+
+This pattern enforces you to write the code which is declarative and pure (immutable).
+
+## Virtual DOM
+HyperApp uses virtual DOM which provides a declarative way of representing the UI of your app. A virtual DOM tree is basically a data structure that consists of nested `vnodes` which describes a DOM tree. Whenever the state of your application changes, a virtual DOM tree is created to update the view. Then it ***diffs*** a `vnode` tree against its previous version and only modifies the DOM where there are changes.
+
+Read more about virtual DOM [here](https://github.com/Matt-Esch/virtual-dom).
+
+## Overview
+
+### Creating a `vnode` tree (hyperscript)
+
+#### `h(tag, properties, children)`
+
+`h()` takes a tag, an optional properties objects and an optional array of children or a single child that is a string. If you pass it an array of children, it will have child nodes.
+
+You can pass deeply nested tags and also components like this
+
+```javascript
+const Component = (data, children) => h("div", data, children)
+
+h(Component, { id: "foo" }, [h(Component, { id: "bar" })])
```

Comment:
Well, you know that the `h` function checks if the tag is a function, so in this case, it will see that MyTag is a function and return calling it.